### DISCUSSION TITLE
Participant information tool fails in querying the participants I created

### DISCUSSION THREAD

**Discussion by TaKO8Ki** [phax/phoss-smp#226] [Q&A] [answered]
Thank you for your response to my previous question. I have one more question. I have successfully registered SMP running locally in SMK and created a service group, but I can't find the participants I registered with the [Participant Information tool](https://peppol.helger.com/public/locale-en_US/menuitem-tools-participant). If you know why this problem happens, I would like to know.

![Screen Shot 2022-08-10 at 17 26 42](https://user-images.githubusercontent.com/41065217/183853598-6804c79d-1b2c-457c-9d6a-b6fcc8f0a56d.png)

Querying the SMP for `iso6523-actorid-upis::9915:mf-test3` failed:

![Screen Shot 2022-08-10 at 17 29 23](https://user-images.githubusercontent.com/41065217/183853946-2e28c1bc-e8fb-4ac4-8087-708a2c253720.png)

**Comment by phax** [comment, marked as answer]
Okay, I just tried myself and your "9915:mf-test3" identifier is indeed not registered in the network.

Sometimes the participant tool delivers invalid results, if you first try to lookup a participant that does not exist, and then you create the Service Group in your SMP and then you try again in the Participant Tool - because the DNS caching is so aggressive, That can take up to 24 hours to resolve. But this is an edge case.

What I need to know are the following information elements:
* What SML have you configured at "SMP Settings"?
* Did you create the Service Group while the "SML was enabled"?
* What certifictae have you configured in your SMP (page certificate information)
* Did you already register your SMP to the SML? If so, what DNS name did you provide?
* Is your SMP already reachable from the Internet? What is the public domain of the SMP (feel free to PM me by email if you don't want to disclose it here)?

**Comment by TaKO8Ki** [comment]
Thank you for replying.

> What SML have you configured at "SMP Settings"?

I have selected SMK (https://acc.edelivery.tech.ec.europa.eu/edelivery-sml).

<img width="998" alt="Screen Shot 2022-08-12 at 14 59 39" src="https://user-images.githubusercontent.com/41065217/184293770-9bfeb7f7-8cc9-421b-b711-a8b135dbcd87.png">

> Did you create the Service Group while the "SML was enabled"?

Yes, I did.

> What certifictae have you configured in your SMP (page certificate information)

I have configured a Peppol pilot v3 certificate.

<img width="974" alt="Screen Shot 2022-08-12 at 15 11 26" src="https://user-images.githubusercontent.com/41065217/184295019-f75964ec-c70e-4b77-a342-cda29bd5d705.png">

> Did you already register your SMP to the SML? If so, what DNS name did you provide?

Yes, I did. The DNS name is `http://a8221a1cdc7e`.

<img width="1006" alt="Screen Shot 2022-08-12 at 15 19 01" src="https://user-images.githubusercontent.com/41065217/184296019-c0db17d2-698f-4f62-9b57-94479b93f061.png">

> Is your SMP already reachable from the Internet? What is the public domain of the SMP (feel free to PM me by email if you don't want to disclose it here)?

No. My SMP was running locally and was not reachable from the Internet.

Probably, Is it a problem that my SMP is not reachable from the Internet?

**Reply by phax** [reply]
Yes, your SMP must be reachable via the Internet - otherwise my testing system (or any other system) cannot access your SMP. Sorry. Peppol runs over the Internet only, and so I guess your hostname `http://a8221a1cdc7e` has limited validity outside your local network ;-)

**Reply by TaKO8Ki** [reply]
Thank you. I understand. What I want to do is just to check if my participants is correctly registered in SMK. This is because I want to make sure I am getting my SMP working well with my test certificate. Is there any way just to do this? Or is the participants not registered successfully in the first place?

**Reply by phax** [reply]
Well the registration of the participant seemed to have worked (otherwise you would have received a red error box in the SMP).

Maybe - and this is just a try - it works, when you access `http://B-c1cfbaaf7801b292d0aa9e6eb1aa461c.iso6523-actorid-upis.acc.edelivery.tech.ec.europa.eu` in your local network. That is the dynamically created DNS name for the participant `iso6523-actorid-upis::9915:mf-test3`.
It is a redirect to `http://a8221a1cdc7e` which might work in your network.

Otherwise you need to make the SMP accessible to the Internet and update the "SML registration" to the new public domain. Then you can test via the Participant tool.

**Reply by TaKO8Ki** [reply]
I'll try it!

**Reply by TaKO8Ki** [reply]
I created a new participant `9915:mf-test5` and accessed its DNS name, but It did not redirect to my SMP and `Check DNS state of participants` page was like the following. Is the reason why the IP address column shows `is not registered in SML` that my SMP is not open to the Internet?

> Otherwise you need to make the SMP accessible to the Internet and update the "SML registration" to the new public domain.

If so, I will try it.

<img width="1227" alt="Screen Shot 2022-08-16 at 16 39 09" src="https://user-images.githubusercontent.com/41065217/184824642-ecae4b09-4541-46a5-b920-0e4388638a85.png">

**Reply by phax** [reply]
The answer is "most likely" ;-) There are so many theoretical way how DNS servers are configured and how they are accessed.
An easy way would also be to try the command `nslookp B-248....tech.ec.europa.eu` and see the name can be resolved on the shell.
Maybe you have an internal DNS server that you can use for querying instead of the default DNS, but that is far outside the range of the SMP :-)

**Reply by TaKO8Ki** [reply]
Thank you! I'll set up a test server and connect it to the access point :)
